api/src/controllers/post/inspection.js

```diff
@@ -0,0 +1,178 @@
+    incomingObj.issuingAgency && (inpsection.issuingAgency = incomingObj.issuingAgency);
+    incomingObj.author && (inpsection.author = incomingObj.author);
+    incomingObj.legislation && (inpsection.legislation = incomingObj.legislation);
+    incomingObj.issuedTo && (inpsection.issuedTo = incomingObj.issuedTo);
+    incomingObj.projectName && (inpsection.projectName = incomingObj.projectName);
+    incomingObj.location && (inpsection.location = incomingObj.location);
+    incomingObj.centroid && (inpsection.centroid = incomingObj.centroid);
+    incomingObj.outcomeStatus && (inpsection.outcomeStatus = incomingObj.outcomeStatus);
+    incomingObj.outcomeDescription && (inpsection.outcomeDescription = incomingObj.outcomeDescription);
+
+    inpsection.dateAdded = new Date();
+    inpsection.publishedBy = args.swagger.params.auth_payload.displayName;
+
+    incomingObj.sourceDateAdded && (inpsection.sourceDateAdded = incomingObj.sourceDateAdded);
+    incomingObj.sourceDateUpdated && (inpsection.sourceDateUpdated = incomingObj.sourceDateUpdated);
+    incomingObj.sourceSystemRef && (inpsection.sourceSystemRef = incomingObj.sourceSystemRef);
+
+    inpsection.read = ['sysadmin'];
+    inpsection.write = ['sysadmin'];
+
+    let savedInspection = null;
+    try {
+        savedInspection = await inpsection.save();
+    } catch (e) {
+        return {
+            status: 'failure',
+            object: inpsection,
+            errorMessage: e
+        }
+    }
+
+    var observables = [];
+    incomingObj.InspectionLNG && observables.push(this.createLNG(args, res, next, incomingObj.InspectionLNG, savedInspection._id));
+    incomingObj.InspectionNRCED && observables.push(this.createNRCED(args, res, next, incomingObj.InspectionLNG, savedInspection._id));
+
+    var flavourRes = null;
+    try {
+        observables.length > 0 && (flavourRes = await Promise.all(observables));
+    } catch (e) {
+        flavourRes = {
+            status: 'failure',
+            object: observables,
+            errorMessage: e
+        }
+    }
+
+    return {
+        status: 'success',
+        object: savedInspection,
+        flavours: flavourRes
+    }
+};
+
+// Example of incomingObj
+/**
+ *  {
+ *      _master: '5e1e7fcd20e4167bcfc3daa7'
+ *      description: 'lng description',
+ *      ...
+ *      addRole: 'public'
+ *  }
+ */
+exports.createLNG = async function (args, res, next, incomingObj, masterId) {
+    var InspectionLNG = mongoose.model('InspectionLNG');
+    var inpsectionLNG = new InspectionLNG();
+
+    inpsectionLNG._schemaName = 'InspectionLNG';
+
+    // We must have a valid master ObjectID to continue.
+    masterId && ObjectId.isValid(masterId) && (inpsectionLNG._master = new ObjectId(masterId));
+    if (!inpsectionLNG._master) {
+        return {
+            status: 'failure',
+            object: incomingObj,
+            errorMessage: 'incomingObj._master was not valid ObjectId'
+        }
+    }
```

Comment:
Minor optimization, but the check for a valid masterId could be the first thing this function checks, before loading the model, etc.